sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SQLAppStatusListener.scala

```diff
@@ -140,7 +146,16 @@ class SQLAppStatusListener(
 
   override def onExecutorMetricsUpdate(event: SparkListenerExecutorMetricsUpdate): Unit = {
     event.accumUpdates.foreach { case (taskId, stageId, attemptId, accumUpdates) =>
-      updateStageMetrics(stageId, attemptId, taskId, accumUpdates, false)
+      updateStageMetrics(stageId, attemptId, taskId, SQLAppStatusListener.UNKNOWN_PARTITION,
+        accumUpdates, false)
+    }
+  }
+
+  override def onTaskStart(event: SparkListenerTaskStart): Unit = {
+    Option(stageMetrics.get(event.stageId)).foreach { stage =>
+      if (stage.attemptId == event.stageAttemptId) {
+        stage.registerTask(event.taskInfo.taskId, event.taskInfo.index)
```

Comment:
Actually, it seems that we need a little comment for this, @vanzin .
By definition, this PR claims the following.
```scala
  /**
    * Mapping of task IDs to the partition index they're computing. Note this may contain more
    * elements than the stage's number of tasks, if speculative execution is on.
    */
   private val taskIndices = new OpenHashMap[Long, Int]()
```

However, we are putting `TaskInfo.Index` which is defined like the following.
```scala
@DeveloperApi
class TaskInfo(
    val taskId: Long,
    /**
     * The index of this task within its task set. Not necessarily the same as the ID of the RDD
     * partition that the task is computing.
     */
    val index: Int,
```

Reply:
See below.

Reply:
Thank you for the explanation. However, simply put, when `taskIndices` maintain a copy of `TaskInfo.taskId` and `TaskInfo.index`, can we call them, `taskId` and `taskIndex`, literally? The variable name is already _task**Indices**_.

Reply:
Man, if that's going to make you drop this discussion, I'll call them "pink unicorns" if that's what you want. The variable name won't help you understand the relationship between partitions and indices and why it's important to track this.